### DISCUSSION TITLE
Magick Image not working in .Net 8 running on Synology NAS

### DISCUSSION THREAD

**Discussion by SeppeVE** [dlemstra/Magick.NET#1590] [Help]
Hey, I was trying to use the Magick.NET package in a project that would run on a Synology NAS. However it would not work in my api, so I made a quick test project [here](https://github.com/SeppeVE/TestMagickImage). But as I run that just as a small test it gives me the same error 
"An unexpected error occurred: The type initializer for 'NativeMagickSettings' threw an exception."
which makes it clear it's not just an API issue.  If anyone knows if there is a way to run this on a DSM 7 NAS it would help me a lot.

**Comment by dlemstra** [comment]
Are you using Visual Studio?

**Reply by SeppeVE** [reply]
Yes, this example and the actual project are both written in Visual Studio. I'm not sure but if you want to suggest something about publishing, I have tried a couple of things.

- Framework-dependent, Portable
- Self-contained, Linux-arm
- Self-contained, Linux-arm64 (which seemed not to run at all, Synology is probably not running 64 bit)

**Reply by dlemstra** [reply]
I have seen problems where the native binaries are not copied to the output folder. You might want to check the output of `ldd` on the native binary with a terminal on your NAS to check what is happening. It might be possible that some of the dependencies are not available on your machine.
